Add obsidian-emoji-titler

HyeonseoNam opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: [https://github.com/HyeonseoNam/obsidian-emoji-titler](https://github.com/HyeonseoNam/obsidian-emoji-titler)

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["fundingUrl": "",](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/manifest.json#L9)
If you don't have one remove this.

[import { basename, join } from 'path';](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L2)
This will break on mobile, as Node.js is not avaiable there.



[const file = app.workspace.getActiveFile();](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L64)
It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future. Instead, you should use the reference to App that you have from your plugin instance.

[type EmojiTitlerSettingsKey = "emoji1" | "emoji2" | "emoji3" | "emoji4" | "emoji5" | "emoji6" | "emoji7" | "emoji8" | "emoji9" | "emoji10";](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L4)
It would be better to use a list so you are not limited to 10 emoji.

[const oldPath = file?.path!;](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L69-L77)
Instead of this you can do `file.getNewPathAfterRename(newName)´.
This method is not exposed, but does the same thing, but device independant.

HyeonseoNam commented:
> ["fundingUrl": "",](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/manifest.json#L9)
> If you don't have one remove this.

removed.

> [import { basename, join } from 'path';](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L2)
> This will break on mobile, as Node.js is not avaiable there.

removed.

> [const file = app.workspace.getActiveFile();](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L64)
> It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future. Instead, you should use the reference to App that you have from your plugin instance.

relaced `app` to `this.app`.

> [type EmojiTitlerSettingsKey = "emoji1" | "emoji2" | "emoji3" | "emoji4" | "emoji5" | "emoji6" | "emoji7" | "emoji8" | "emoji9" | "emoji10";](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L4)
> It would be better to use a list so you are not limited to 10 emoji.

I attempted to handle more than 10 emojis, and also tried to add a feature to delete them. But I got stuck with deleting the existing command (that registered by `this.addCommand`). Could I handle this in the later update?


> [const oldPath = file?.path!;](https://github.com/HyeonseoNam/obsidian-emoji-titler/blob/44c3dd096ec07dba29727a4ffe8c4bedda72e4bd/main.ts#L69-L77)
> Instead of this you can do `file.getNewPathAfterRename(newName)´.
> This method is not exposed, but does the same thing, but device independant.

replaced to `file.getNewPathAfterRename(newName)`.
